spaghetti/network.py

```diff
@@ -130,9 +136,12 @@ def _extractnetwork(self):
         Used internally, to extract a network from a polyline shapefile.
         """
         nodecount = 0
-        shps = ps.open(self.in_shp)
+        if isinstance(self.in_shp, str):
+            shps = ps.open(self.in_shp)
+        else:
+            shps = self.in_shp["geometry"]
```

Comment:
this needs to be `self.in_shp.geometry`, since the geometry column [can be named whatever you like](https://github.com/geopandas/geopandas/issues/755).